Documentation: Add supported data types

MKKnorr opened the pull request:
Added supported data types to the documentation, plus some minor fixes

MKKnorr committed:
Documentation: Add supported data types

umfranzw commented on docs/data-type-support.rst:
```diff
@@ -0,0 +1,86 @@
+      - ✅
+      - ✅
+    *
+      - :code:`half` [1]_
```
I have a quick question about the footnote here. In the rocPRIM data type documentation table, I noticed that we currently don't support calling rocprim::device_reduce_by_key with the rocprim::half type. Does that mean we also don't support thrust::reduce_by_key with the half type on the device? If so, it might be nice to add to the footnote.

MKKnorr replied:
I think `rocprim::device_reduce_by_key` actually fails due to verification problems on the host side. As the block/thread level functions work, I would expect it to also work in thrust. I can double check, but the verification would most likely also fail due to the missing functionality of `half` on the host

neon60 commented:
@MKKnorr bfloat16?

MKKnorr committed:
Update docs: supported data types

Add bfloat16
Use type name conventions as given by the ROCm docs data type landing page

MKKnorr commented:
> @MKKnorr bfloat16?

Added to the table

neon60 commented on docs/data-type-support.rst:
```diff
@@ -0,0 +1,62 @@
+rocThrust and Thrust support custom, user-defined types, if they provide the interface required by the used functions.
+
+.. rubric:: Footnotes
+.. [1] Limited support on the host side.
```
Can you open this up in two sentences?

MKKnorr replied:
Added a short explanatoin

neon60 commented:
@stanleytsang-amd Can you check the PR?

umfranzw reviewed (approved):
Looks good to me. The gfx1030 check failure is unrelated to this change.

MKKnorr commented:
> Looks good to me. The gfx1030 check failure is unrelated to this change.

Thanks, please merge if nothing else is blocking this
